src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```
@@ -0,0 +1,126 @@
+        if (!simulatorOutputFileOpt->empty()){
+            simulatorOutputFileBuffer.open(simulatorOutputFile, std::ios::out);
+            std::ostream simulatorOutputFileStream(&simulatorOutputFileBuffer);
+            // TODO: Call into the QIR runtime API to redirect the output of the simulator.
+            simulatorOutputStream = &simulatorOutputFileStream;
+        }
+
+        // Redirect the Q# opertion output from std::cout if the --operation-output option is present.
+        std::ostream* operationOutputStream = &std::cout;
+        std::filebuf operationOutputFileBuffer;
+        if (!operationOutputFileOpt->empty()){
+            operationOutputFileBuffer.open(operationOutputFile, std::ios::out);
+            std::ostream operationOutputFileStream(&operationOutputFileBuffer);
+            operationOutputStream = &operationOutputFileStream;
+        }
+
+        // TODO: Remove this after the Message Q# function is integrated into the QIR runtime.
+        (*operationOutputStream) << "SIMULATOR OUTPUT\n----------------" << std::endl;
+
+        // Start simulation.
+        std::unique_ptr<ISimulator> sim = CreateFullstateSimulator();
+        QirContextScope qirctx(sim.get(), false /*trackAllocatedObjects*/);
+        Quantum__StandaloneSupportedInputs__ExerciseInputs__body(anInt, aDouble);
+        simulatorOutputStream->flush();
+
+        // This is for visualization purposes only and should not be part of the generated C++ code.
+        // N.B. For examples on how other return values are shown, look at other samples.
+        (*operationOutputStream) << "\nOPERATION OUTPUT\n----------------" << std::endl;
+        operationOutputStream->flush();
+
+        // Close opened file buffers;
+        if (!simulatorOutputFileOpt->empty()){
+            simulatorOutputFileBuffer.close();
+        }
+
+        if (!operationOutputFileOpt->empty()){
+            operationOutputFileBuffer.close();
+        }
+    } catch(...) {
```

Why not let it throw and terminate with, hopefully, a more specific message?

---

I removed the global try-catch.